django/db/backends/mysql/introspection.py

```diff
@@ -211,42 +211,37 @@ def get_constraints(self, cursor, table_name):
         # Get the actual constraint names and columns
         name_query = """
             SELECT kc.`constraint_name`, kc.`column_name`,
-                kc.`referenced_table_name`, kc.`referenced_column_name`
-            FROM information_schema.key_column_usage AS kc
+                kc.`referenced_table_name`, kc.`referenced_column_name`,
+                c.`constraint_type`
+            FROM
+                information_schema.key_column_usage AS kc,
+                information_schema.table_constraints AS c
             WHERE
                 kc.table_schema = DATABASE() AND
+                c.table_schema = kc.table_schema AND
+                c.constraint_name = kc.constraint_name AND
                 kc.table_name = %s
             ORDER BY kc.`ordinal_position`
         """
         cursor.execute(name_query, [table_name])
-        for constraint, column, ref_table, ref_column in cursor.fetchall():
+        for constraint, column, ref_table, ref_column, kind in cursor.fetchall():
             if constraint not in constraints:
                 constraints[constraint] = {
                     'columns': OrderedSet(),
-                    'primary_key': False,
-                    'unique': False,
+                    'primary_key': kind == 'PRIMARY KEY',
+                    'unique': kind in {'PRIMARY KEY', 'UNIQUE'},
                     'index': False,
                     'check': False,
                     'foreign_key': (ref_table, ref_column) if ref_column else None,
                 }
                 if self.connection.features.supports_index_column_ordering:
                     constraints[constraint]['orders'] = []
-            constraints[constraint]['columns'].add(column)
-        # Now get the constraint types
-        type_query = """
-            SELECT c.constraint_name, c.constraint_type
-            FROM information_schema.table_constraints AS c
-            WHERE
-                c.table_schema = DATABASE() AND
-                c.table_name = %s
-        """
-        cursor.execute(type_query, [table_name])
-        for constraint, kind in cursor.fetchall():
-            if kind.lower() == "primary key":
+            elif kind == 'PRIMARY KEY':
```

Comment:
Unfortunately unnamed unique and check columns constraints have the same name as a column (see 579909a13feb958f75d1d77e9995cef8f2b9c2a3) so we need to keep `elif` blocks.